content/6_Advanced/Eulers_Formula.mdx

```diff
@@ -35,67 +231,67 @@ int hsh(int a, int b) { return N * a + b; }
 const int xd[4] = {1, 0, -1, 0}, yd[4] = {0, 1, 0, -1};
 
 template <int SZ> struct DSU {
-	int par[SZ], sz[SZ], measure[SZ];
-	vi comp[SZ];
-	DSU() { F0R(i, SZ) par[i] = i, sz[i] = 1, measure[i] = 1; }
-	bool valid(int b, int c) { return b >= 0 && b < N && c >= 0 && c < N; }
-	bool ok(int a, int b, int c) {
-		if (!valid(b, c)) return 0;
-		return par[hsh(b, c)] == a;
-	}
-	void addPoint(int x, pi t) {
-		par[hsh(t.f, t.s)] = x;
-		measure[x]++;
-		F0R(i, 4) {
-			if (ok(x, t.f + xd[i], t.s + yd[i])) {
-				measure[x]--;
-				int j = (i + 1) % 4;
-				if (ok(x, t.f + xd[j], t.s + yd[j]) &&
-				    ok(x, t.f + xd[j] + xd[i], t.s + yd[j] + yd[i]))
-					measure[x]++;
-			}
-		}
-		comp[x].pb(hsh(t.f, t.s));
-	}
-	void unite(pi x, pi y) {  // union-by-rank
-		int X = hsh(x.f, x.s), Y = hsh(y.f, y.s);
-		if (par[X] == par[Y]) return;
-		X = par[X], Y = par[Y];
-		if (sz(comp[X]) < sz(comp[Y])) swap(X, Y);
-		trav(t, comp[Y]) addPoint(X, {t / N, t % N});
-		comp[Y].clear();
-	}
+    int par[SZ], sz[SZ], measure[SZ];
+    vi comp[SZ];
+    DSU() { F0R(i, SZ) par[i] = i, sz[i] = 1, measure[i] = 1; }
+    bool valid(int b, int c) { return b >= 0 && b < N && c >= 0 && c < N; }
```

Comment:
you didn't touch this code did you

Reply:
I didn't. I thought my job was to just explain it.

Reply:
Just for the second example